### ISSUE TITLE
wrong acc, loss return in `one_iter_of_cifar10_training`

### ISSUE THREAD

**Issue by tiendung** [dlsyscourse/hw4#2]
https://github.com/dlsyscourse/hw4/blob/d6ed0352bab6c0f34a329ecc15984cd1dc079ed3/tests/test_conv.py#L489

```py
    for batch in dataloader:
        opt.reset_grad()
        X, y = batch
        X,y = ndl.Tensor(X, device=device), ndl.Tensor(y, device=device)
        out = model(X)
        correct += np.sum(np.argmax(out.numpy(), axis=1) == y.numpy())
        loss = loss_fn(out, y)
        total_loss += loss.data.numpy() * y.shape[0]
        loss.backward()
        opt.step()
        if i >= niter:
            break
        i += 1
    return correct/(y.shape[0]*niter), total_loss/(y.shape[0]*niter)
```
`correct/(y.shape[0]*niter), total_loss/(y.shape[0]*niter)` only correct if the size of the last batch is equal to the others. It's maybe not true for all dataset.
